src/Symfony/Component/Messenger/DataCollector/MessengerDataCollector.php

```diff
@@ -14,21 +14,34 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\DataCollector\DataCollector;
-use Symfony\Component\Messenger\MiddlewareInterface;
+use Symfony\Component\Messenger\TraceableMessageBus;
 
 /**
  * @author Samuel Roze <samuel.roze@gmail.com>
  *
  * @experimental in 4.1
  */
-class MessengerDataCollector extends DataCollector implements MiddlewareInterface
+class MessengerDataCollector extends DataCollector
 {
+    private $traceableBuses = array();
+
+    public function registerBus(string $name, TraceableMessageBus $bus)
```

Comment:
Why not injected in the constructor?

Reply:
This is to match with what already exists like the [`CacheDataCollector`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/DataCollector/CacheDataCollector.php#L36-L39) for example.